provider/segmenter/src/transform.rs

```diff
@@ -0,0 +1,735 @@
+        _ => {
+            panic!("Invalid property name")
+        }
+    }
+}
+
+fn get_line_segmenter_value_from_name(name: &str) -> LineBreak {
+    match name {
+        "AI" => LineBreak::Ambiguous,
+        "AL" => LineBreak::Alphabetic,
+        "B2" => LineBreak::BreakBoth,
+        "BA" => LineBreak::BreakAfter,
+        "BB" => LineBreak::BreakBefore,
+        "BK" => LineBreak::MandatoryBreak,
+        "CB" => LineBreak::ContingentBreak,
+        "CJ" => LineBreak::ConditionalJapaneseStarter,
+        "CL" => LineBreak::ClosePunctuation,
+        "CM" => LineBreak::CombiningMark,
+        "CP" => LineBreak::CloseParenthesis,
+        "CR" => LineBreak::CarriageReturn,
+        "EB" => LineBreak::EBase,
+        "EM" => LineBreak::EModifier,
+        "EX" => LineBreak::Exclamation,
+        "GL" => LineBreak::Glue,
+        "H2" => LineBreak::H2,
+        "H3" => LineBreak::H3,
+        "HL" => LineBreak::HebrewLetter,
+        "HY" => LineBreak::Hyphen,
+        "ID" => LineBreak::Ideographic,
+        "IN" => LineBreak::Inseparable,
+        "IS" => LineBreak::InfixNumeric,
+        "JL" => LineBreak::JL,
+        "JT" => LineBreak::JT,
+        "JV" => LineBreak::JV,
+        "LF" => LineBreak::LineFeed,
+        "NL" => LineBreak::NextLine,
+        "NS" => LineBreak::Nonstarter,
+        "NU" => LineBreak::Numeric,
+        "OP" => LineBreak::OpenPunctuation,
+        "PO" => LineBreak::PostfixNumeric,
+        "PR" => LineBreak::PrefixNumeric,
+        "QU" => LineBreak::Quotation,
+        "RI" => LineBreak::RegionalIndicator,
+        "SA" => LineBreak::ComplexContext,
+        "SG" => LineBreak::Surrogate,
+        "SP" => LineBreak::Space,
+        "SY" => LineBreak::BreakSymbols,
+        "WJ" => LineBreak::WordJoiner,
+        "XX" => LineBreak::Unknown,
+        "ZW" => LineBreak::ZWSpace,
+        "ZWJ" => LineBreak::ZWJ,
+        _ => {
+            panic!("Invalid property name: {}", name)
+        }
+    }
+}
+
+fn is_cjk_fullwidth(eaw: &CodePointTrie<EastAsianWidth>, codepoint: u32) -> bool {
+    matches!(
+        eaw.get(codepoint),
+        EastAsianWidth::Ambiguous | EastAsianWidth::Fullwidth | EastAsianWidth::Wide
+    )
+}
+
+/// A data provider reading from segmenter rule files.
+#[derive(Debug)]
+pub struct SegmenterRuleProvider {
+    /// The path to the segmenter rule break TOML files.
+    data_root: PathBuf,
+
+    /// A provider for the Unicode properties required to generate the segmenter runtime data.
+    provider: icu_provider_fs::FsDataProvider,
+}
+
+impl SegmenterRuleProvider {
+    /// Create a new [`Self`] given a filesystem directory. See [module-level documentation](crate)
+    /// for its usage.
+    pub fn try_new<P: Into<PathBuf>>(data_root: P) -> Result<Self, DataError> {
+        let data_root = data_root.into();
+        data_root.metadata()?; // Check if data_root exists.
+        Ok(Self {
+            data_root,
+            provider: icu_testdata::get_provider(),
```

Comment:
Note: This is the spot where you should wire in the appropriate provider from icu_provider_uprops. You should pass the uprops root to the constructor.